R/create_mofa_from_mae.R

```diff
@@ -0,0 +1,106 @@
+#' filled in the corresponding slots
+#'
+#' @name create_mofa_from_mae
+#' 
+#' @examples
+#' # Load package and import dataset
+#' library(mia)
+#' data("HintikkaXOData", package = "mia")
+#' mae <- HintikkaXOData
+#' 
+#' # Prepare basic model with selected assays
+#' prep_model <- mofa2(mae, assay.types = c("counts", "nmr", "signals"))
+#' 
+#' # Specify grouping variable and extract metadata
+#' prep_model <- mofa2(mae, assay.types = c("counts", "nmr", "signals"),
+#'                     groups = "Diet", extract_metadata = TRUE)
+#'
+#' # Modify MOFA options with corresponding arguments
+#' prep_model <- mofa2(mae, assay.types = c("counts", "nmr", "signals"),
+#'                     num_factors = 5, stochastic = TRUE)
+NULL
+
+#' @rdname create_mofa_from_mae
+#' @export
+setGeneric("mofa2", signature = c("mae"),
+           function(mae, ...) standardGeneric("mofa2")
+)
+
+#' @rdname create_mofa_from_mae
+#' @export
+setMethod("mofa2", signature = c(mae = "MultiAssayExperiment"),
+          function(mae, assay.types = rep("counts", length(mae)),
+                   groups = NULL, extract_metadata = FALSE, ...){
+            # Select assays of each experiment for MOFA
+            mae <- .select_assays(mae, assay.types)
+            # Create MOFA from selected experiments
+            mdl <- create_mofa_from_MultiAssayExperiment(
+              mae,
+              groups = groups,
+              extract_metadata = extract_metadata
+            )
+            # Make a list of arguments for MOFA
+            mofa_args <- list(
+              object = mdl,
+              data_options = .set_opts(get_default_data_options(mdl), ...),
+              model_options = .set_opts(get_default_model_options(mdl), ...),
+              training_options = .set_opts(get_default_training_options(mdl), ...),
+              mefisto_options = .set_opts(get_default_mefisto_options(mdl), ...)
+            )
+            # Add stochastic options if stochastic is turned on
+            if ( mofa_args[["training_options"]][["stochastic"]] ){
+              mofa_args[["stochastic_options"]] <- .set_opts(get_default_stochastic_options(mdl), ...)
+            }
+            # Prepare MOFA
+            prep_mdl <- do.call("prepare_mofa", mofa_args)
+            return(prep_mdl)
+          }
+)
+
+########################## HELP FUNCTIONS ##########################
+
+# Select assays to be included in MOFA
+#' @importFrom MultiAssayExperiment experiments
+#' @importFrom SummarizedExperiment assay assays assayNames
+.select_assays <- function(mae, assay.types) {
+  # Give corresponding experiment names to assay.types
+  names(assay.types) <- names(experiments(mae))
+  # For every experiment in MAE
+  for ( exp in names(experiments(mae)) ){
+    # Keep only selected assay.type from a given experiment
+    assays(mae[[exp]]) <- list(assay(mae[[exp]], assay.types[[exp]]))
+    # Update assay names
+    assayNames(mae[[exp]]) <- assay.types[[exp]]
+  }
+  return(mae)
+}
```

Comment:
These can lead to an errors that might be hard for user to solve:

1. MAE can have also other datatypes than SEs. The object just has to have rows and columns.
2. What if the length of assay.types does not match with length of experiments?